src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1480,7 +1478,34 @@ public void SaveAs(string path, bool isBackup = false)
             }
         }
 
-        
+        /// <summary>
+        /// Save the current workspace to a specific file path. If the file path is null or empty, an
+        /// exception is written to the console.
+        /// </summary>
+        /// <param name="id">Indicate the id of target workspace view model instead of defaulting to 
+        /// current workspace view model. This is critical in crash cases.</param>
+        /// <param name="path">The path to the file.</param>
+        /// <param name="isBackup">Indicates if an automated backup save has called this function.</param>
+        /// <exception cref="IOException"></exception>
+        /// <exception cref="UnauthorizedAccessException"></exception>
+        public void SaveAs(Guid id, string path, bool isBackup = false)
```

Comment:
How about making this new method private. We can always make it part of the public API if needed later.